### ISSUE TITLE
Remove deprecated FTPClient.changeDirectory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6759]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6759|
|Type|task|
|Created|2013-09-23 20:31:43Z|
|Branch|https://github.com/twisted/twisted/tree/remove-changeDirectory-6759|

`FTPClient.changeDirectory` in `twisted.protocols.ftp` was deprecated in version 8.2 and can be removed now.

Attachments:

* [6759.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f01/f010b55b1c66c02310b6d170802b6629de98aae2/b3b361f3eed9c6cb5f6941dfe707609b72fdfa0d.patch) (5063 bytes) - added by ewong on 2013-11-05 02:47:33Z - 
* [6759.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f01/f010b55b1c66c02310b6d170802b6629de98aae2/0921a0ad429b04733d419627ddd68cad25e6e59a.patch) (750 bytes) - added by ewong on 2013-11-05 13:10:14Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6759 6759
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_remove_changeDirectory_6759 branches/remove-changeDirectory-6759
branch_author__lvh lvh
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1379968303000000 1379968303000000
changetime__1383658752000000 1383658752000000
version__None None
owner__lvh lvh
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong set owner to ewong|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_4"></a>|@lvh commented|
|-|-|

(In [40576]) Branching to 'remove-changeDirectory-6759'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_5"></a>|@lvh set owner to ewong|
|-|-|

Hi ewong!


Thank you for your patch. There are some new pyflakes problems:

```
twisted/protocols/ftp.py:17: 'warnings' imported but unused
twisted/test/test_ftp.py:16: 'util' imported but unused
twisted/test/test_ftp.py:796: redefinition of function 'checkDownload' from line 790
twisted/test/test_ftp.py:817: redefinition of function 'checkDownload' from line 805
```

The first two should be obvious: the modules are no longer using ```warnings``` or ```util``` anywhere else, so please remove those. The latter two aren't real problems since they're callback functions, so you don't have to fix those: however, you should file a bug with pyflakes saying that redefinition errors should be suppressed if a function is used before being redefined :)

As expected there are no failures on the buildbots outside of that.

Please address the things I mentioned and supply a new patch (against branches/remove-changeDirectory-6759 please :)), and I'll be happy to merge it for you :)

Thanks again for your contribution

lvh

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|ewong removed owner|
|-|-|

Updated the patch with the removal of the two unused imports.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_7"></a>|@lvh commented|
|-|-|

(Please add the review keyword, otherwise people won't see it in the review queue :) I happened to see it because I'm subscribed.)

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_10"></a>|@lvh commented|
|-|-|

Hi again :)

Current new pyflakes failures:

```
twisted/protocols/ftp.py:21: redefinition of unused 'grp' from line 19
twisted/protocols/ftp.py:21: redefinition of unused 'pwd' from line 19
twisted/test/test_ftp.py:796: redefinition of function 'checkDownload' from line 790
twisted/test/test_ftp.py:817: redefinition of function 'checkDownload' from line 805
```

Those redefinitions are all spurious (they're in a try/except), so this looks good to me! Thanks again for your contribution!

I'm checking that the current WinXP failures are already accounted for in different tickets, but I'm pretty sure they're not *this* ticket, so I think this is good to merge :)

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_11"></a>|@lvh commented|
|-|-|

Good to go; there's plenty of tickets related to that infernal testManyProcesses problem on Windows, so that isn't a blocker for this ticket :)

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_12"></a>|@lvh set status to `closed`|
|-|-|

(In [40580]) Merge remove-changeDirectory-6759: Remove deprecated FTPClient.changeDirectory

Author: ewong
Reviewer: lvh
Fixes: [#6759](https://github.com/twisted/twisted/issues/6759)

FTPClient.changeDirectory in twisted.protocols.ftp was deprecated in version 8.2 and is removed now.
